### DISCUSSION TITLE
Getting nicer symbolic representations

### DISCUSSION THREAD

**Discussion by RicardRC** [MilesCranmer/PySR#788] [Q&A] [answered]
Hello! Thanks for the package.
Is there any way of obtaining the correct parameter names from the report? I would like to pass the array of names as third parameter to node_to_symbolic(...) to get recognizable parameters after processing the symbolic version further. Currently I'm getting x1, x2, etc. I've seen that printing the whole report, there's a variable_names vector. But I can't find how to access it as a julia newbie.

`equations = Expression{Float64, DynamicExpressions.NodeModule.Node{Float64}, @NamedTuple{operators::DynamicExpressions.OperatorEnumModule.OperatorEnum{Tuple{typeof(+), typeof(-), typeof(*), typeof(/), typeof(safe_pow), typeof(rem), typeof(/)}, Tuple{typeof(square), typeof(safe_sqrt), typeof(neg), typeof(exp), typeof(safe_sqrt), typeof(safe_log), typeof(safe_log2), typeof(safe_log10), typeof(inv), typeof(abs), typeof(sign)}}, variable_names::Vector{String}}}[...]`

**Comment by MilesCranmer** [comment, marked as answer]
Oh you can get the variables names with

```julia
using SymbolicRegression: get_variable_names
eq = equations[end]
vars = get_variable_names(eq)
```

This is the “proper” generic way, but you can also just do

```julia
eq.metadata.variable_names
```

**Reply by RicardRC** [reply]
- using get_variable_names(eq) gets me `ERROR: UndefVarError: `get_variable_names` not defined in `Main``

- Using `node_to_symbolic(eq, model, eq.metadata.variable_names)` gets me a "Please load SymbolicUtils" (which is loaded, so something else is wrong), but at least the displayed result I get is a vector of Symbol names, so I guess now there's only figuring out what is the exact signature of node_to_symbolic.

**Reply by RicardRC** [reply]
Yeah, it was my unfamiliarity with julia. Luckily you have test_symbolic_utils.jl in here which helped me find the proper way.
```
eqs = node_to_symbolic(eq, model, variable_names=eq.metadata.variable_names)
display(eqs)
```
This worked fine! Thank you again!

**Comment by MilesCranmer** [comment]
Although, the `node_to_symbolic` should automatically extract the variable names. If not we should fix it

**Reply by RicardRC** [reply]
It currently does not, at least not to me. Maybe I'm doing something wrong?

**Reply by MilesCranmer** [reply]
It could be I just forgot to merge that feature! I'll try to take a look.

**Comment by MilesCranmer** [comment]
Ok should be fixed by this: https://github.com/SymbolicML/DynamicExpressions.jl/pull/117
